internal/sre/server.go

```diff
@@ -0,0 +1,1590 @@
+package sre
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/gorilla/websocket"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kagent-dev/khook/api/v1alpha2"
+	"github.com/kagent-dev/khook/internal/interfaces"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Alert represents an alert for SRE-IDE
+type Alert struct {
+	ID                string    `json:"id"`
+	HookName          string    `json:"hookName"`
+	Namespace         string    `json:"namespace"`
+	EventType         string    `json:"eventType"`
+	ResourceName      string    `json:"resourceName"`
+	Severity          string    `json:"severity"`
+	Status            string    `json:"status"`
+	Timestamp         time.Time `json:"timestamp"`
+	FirstSeen         string    `json:"firstSeen"`
+	LastSeen          string    `json:"lastSeen"`
+	Message           string    `json:"message"`
+	AgentID           string    `json:"agentId"`
+	SessionID         *string   `json:"sessionId,omitempty"`
+	TaskID            *string   `json:"taskId,omitempty"`
+	RemediationStatus *string   `json:"remediationStatus,omitempty"`
+}
+
+// AlertSummary represents alert statistics
+type AlertSummary struct {
+	Total        int            `json:"total"`
+	Firing       int            `json:"firing"`
+	Resolved     int            `json:"resolved"`
+	Acknowledged int            `json:"acknowledged"`
+	BySeverity   map[string]int `json:"bySeverity"`
+	ByEventType  map[string]int `json:"byEventType"`
+}
```

Comment:
Should we rely on a db for this similar to kagent. We can use `sqlite` by default, but rely on the same Postgres instance of `kagent` if it's available.